model/src/resource/mod.rs

```diff
@@ -31,6 +32,12 @@ pub enum Transport {
     RtpUnicast,
     RtpMulticast,
     Dash,
+    Http,
+}
+
+pub trait Registerable {
```

Comment:
Used this trait to make registration / posting / removal of resources generic. With the old approach i would have had to write like 24 functions. Ideally more traits / generic implementations would have been used in here, but this was just the most egregious one